static/js/people.js

```diff
@@ -1028,7 +1028,11 @@ exports.my_custom_profile_data = function (field_id) {
 };
 
 exports.get_custom_profile_data = function (user_id, field_id) {
-    return people_by_user_id_dict.get(user_id).profile_data[field_id];
+    var profile_data = people_by_user_id_dict.get(user_id).profile_data;
+    if (profile_data === undefined) {
+        return null;
+    }
```

Comment:
I think it might be better to write this as as a case for bots, specifically?

I'm not really sure what semantics you want here; why return null rather than the old code's undefined here?

Reply:
I could have just as well return _undefined_ here. But the important thing that we're now doing here is that we're first checking if a profile_data object exists before we try to retrieve an element of it.

before, we'd try to directly return `people_by_user_id_dict.get(user_id).profile_data[field_id]` and if profile_data did not exist an error would be raised (the first gif shows this error) because we can't do a get-by-key operation on an undefined/null type.

Reply:
> I think it might be better to write this as as a case for bots, specifically?

I guess that in a way we're doing that with this check since I think that only bots will have a missing profile_data object (though I'm not 100% sure).